sourcetool/pkg/sourcetool/tool.go

```diff
@@ -38,3 +52,109 @@ func (t *Tool) GetRepoControls(funcs ...ooFn) (slsa.Controls, error) {
 
 	return t.impl.GetActiveControls(&opts)
 }
+
+// OnboardRepository configures a repository to set up the required controls
+// to meet SLSA Source L3.
+func (t *Tool) OnboardRepository(funcs ...ooFn) error {
+	opts := t.Options
+	for _, f := range funcs {
+		if err := f(&opts); err != nil {
+			return err
+		}
+	}
+
+	if err := t.impl.EnsureDefaults(&opts); err != nil {
```

Comment:
optional/future: Instead of redoing the logic here call ConfigureControls with all the configs?